app/assets/javascripts/application.js

```diff
@@ -14,3 +14,106 @@
 //= require jquery_ujs
 
 //= require_tree .
+$(document).ready(function() {
+
+  var x = document.getElementById("demo");
+  var minUpdateUserDistance = 0.01 //km
+  var userId = $("div[style='display:none']").attr('id')
+  var firebaseRef = new Firebase("https://blinding-fire-43.firebaseio.com/");
+  var geo_options = {
+  enableHighAccuracy: true,
+  maximumAge        : 30000,
+  timeout           : 27000
+  };
+
+  var currentdate = new Date();
+
+
+
+
+  // Create a GeoFire index
+  var geoFire = new GeoFire(firebaseRef);
+
+
+  function getLocation() {
+      if (navigator.geolocation) {
+          navigator.geolocation.watchPosition(checkGeoFire, null, geo_options);
+      } else {
+          x.innerHTML = "Geolocation is not supported by this browser.";
+      }
+  }
+
+
+  function checkGeoFire(position) {
+    // firebaseRef.push(userId)
+
+    geoFire.get(userId).then(function(location) {
+      if (location === null) {
+        geoFire.set(userId, [position.coords.latitude, position.coords.longitude]);
+        firebaseRef.child(userId).update({'datetime': currentdate});
+        console.log('No key in database so we set the initial position')
+      }
+      else {
+        if (GeoFire.distance([position.coords.latitude, position.coords.longitude], location) > minUpdateUserDistance ) {
+          geoFire.set(userId, [position.coords.latitude, position.coords.longitude]);
+          console.log('checked the key and we moved enough to update database')
+        }
+        }
+      }, function(error) {
+        console.log("Error: " + error);
+      }
+    );
+    firebaseRef.child(userId).update({'datetime': currentdate});
+  }
+
+
+
+getLocation()
+
+  var audio;
+  var playlist;
+  var tracks;
+  var current;
+
+  init();
+  function init(){
+      current = 0;
+      audio = $('audio');
+
+
+      playlist = $('#playlist');
+      tracks = playlist.find('span a');
+      firstTrackSrc = tracks.first().attr('href')
```

Comment:
You should have new lines between every 3 or so lines of code.  Think of chunking your code into related paragraphs.